hamiltonian_generative_network.py

```diff
@@ -95,38 +98,48 @@ def forward(self, rollout_batch, n_steps=None):
             prediction.append_reconstruction(x_reconstructed)
         return prediction
 
-    def fit(self, rollouts):
+    def fit(self, rollouts, variational=True):
         """Perform a training step with the given rollouts batch.
 
+        TODO: Move the whole fit() code inside forward if adding hooks to the training, as direct
```

Comment:
I'd prefer them to be separate so I can forward test rollouts. And try to extend the evolution in time.

Reply:
Yes yes, it's okay like this for now. But if we will ever use hooks we cannot directly call the `forward()` of the model I think (see [this discussion](https://discuss.pytorch.org/t/any-different-between-model-input-and-model-forward-input/3690)).